interop-testing/src/main/java/io/grpc/testing/integration/AbstractInteropTest.java

```diff
@@ -1966,4 +2013,53 @@ private static ServerInterceptor recordContextInterceptor(
       }
     };
   }
+
+  /**
+   * A marshaller that record input and output sizes.
+   */
+  private static final class ByteSizeMarshaller<T> implements MethodDescriptor.Marshaller<T> {
+
+    private final MethodDescriptor.Marshaller<T> delegate;
+    volatile int lastOutSize;
+    volatile int lastInSize;
+
+    ByteSizeMarshaller(MethodDescriptor.Marshaller<T> delegate) {
+      this.delegate = delegate;
+    }
+
+    @Override
+    public InputStream stream(T value) {
+      InputStream is = delegate.stream(value);
+      byte[] dst = new byte[16 * 1024 * 1024];
```

Comment:
Use Guava (like CountingInputStream) or our own IoUtils.toByteArray?

(Note that the volatiles aren't necessary since the RPC has completed and the response was returned.)

Reply:
CountingInputStream is beta, but it since it's a testing package it should be fine.

Reply:
The ideal thin would be ByteStreams.copy, but thats beta too.  I can use it if its okay.

Reply:
It's fine to use `@Beta` APIs in tests.

Reply:
Done.